### PR TITLE
Updated the file with latest spec file.

### PR DESCRIPTION
**Pull request by bhupendra-vaishnav**
Updated the file with latest spec file to fix the issue # [352](https://github.com/devcontainers/devcontainers.github.io/issues/352)

### CODE REVIEW

File: _implementors/json_reference.md [hunk 1]
```diff
@@ -8,13 +8,13 @@ index: 4
...
 
-Metadata properties marked with a 🏷️️ can be stored in the `devcontainer.metadata` **[container image label](/implementors/reference/#labels)** in addition to `devcontainer.json`. This label can contain an array of json snippets that will be automatically merged with `devcontainer.json` contents (if any) when a container is created.
+Metadata properties marked with a 🏷️ can be stored in the `devcontainer.metadata` **[container image label](/implementors/reference/#labels)** in addition to `devcontainer.json`. This label can contain an array of json snippets that will be automatically merged with `devcontainer.json` contents (if any) when a container is created.
 
 ## <a href="#general-properties" name="general-properties" class="anchor"> General devcontainer.json properties </a>
 
 | Property | Type  | Description |
 |:------------------|:------------|:------------|
-| `name` | string | A name for the dev container displayed in the UI. |
+| `name` | string | A name for the dev container displayed in the UI |
```
Comment (samruddhikhandale) [inline _implementors/json_reference.md]: revert?

File: _implementors/json_reference.md [hunk 2]
```diff
@@ -48,10 +48,11 @@ The focus of `devcontainer.json` is to describe how to enrich a container for th
...
+| `appPort` | integer,<br>string,<br>array |  In most cases, we recommend using the new [forwardPorts property](#general-devcontainerjson-properties). This property accepts a port or array of ports that should be published locally when the container is running. Unlike `forwardPorts`, your application may need to listen on all interfaces (`0.0.0.0`) not just `localhost` for it to be available externally. Defaults to `[]`.<br>Learn more about publishing vs forwarding ports [here](#publishing-vs-forwarding-ports).<br>Note that the array syntax will execute the command without a shell. You can [learn more](#formatting-string-vs-array-properties) about formatting string vs array properties. |
```
Comment (samruddhikhandale) [inline _implementors/json_reference.md]: Why do we need to update to `#general-devcontainerjson-properties` ? I don't see that section present

File: _implementors/json_reference.md [hunk 3]
```diff
@@ -76,31 +77,29 @@ When creating or working with a dev container, you may need different commands t
 
 | Property | Type  | Description |
 |:------------------|:------------|:------------|
-| `initializeCommand` | string,<br>array,<br>object | A command string or list of command arguments to run on the **host machine** before the container is created.<br /><br /> ⚠️ The command is run wherever the source code is located on the host. For cloud services, this is in the cloud.<br><br>Note that the array syntax will execute the command without a shell. You can [learn more](#formatting-string-vs-array-properties) about formatting string vs array vs object properties. |
```
Comment (samruddhikhandale) [inline _implementors/json_reference.md]: Let's keep this extra `<br>`. It's good for visibility.

File: _implementors/json_reference.md [hunk 4]
```diff
@@ -76,31 +77,29 @@ When creating or working with a dev container, you may need different commands t
...
 {: .table .table-bordered .table-responsive}
 
-For each command property, if the value is a single string, it will be run in `/bin/sh`. Use `&&` in a string to execute multiple commands. For example, `"yarn install"` or `"apt-get update && apt-get install -y curl"`. The array syntax `["yarn", "install"]` will invoke the command (in this case `yarn`) directly without using a shell. Each fires after your source code has been mounted, so you can also run shell scripts from your source tree. For example: `bash scripts/install-dev-tools.sh`.
+For each command property, if the value is a single string, it will be run in `/bin/sh`. Use `&&` in a string to execute multiple commands. For example, `"yarn install"` or `"apt-get update && apt-get install -y curl"`. The array syntax `["yarn", "install"]` will invoke the command (in this case `yarn`) directly without using a shell. Each fires after your source code has been mounted, so you can also run shell scripts from your source tree. For example: `bash scripts/install-dev-tools.sh`
```
Comment (samruddhikhandale) [inline _implementors/json_reference.md]: Revert?

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: _implementors/json_reference.md (hunk 1 above)
Comment (bhupendra-vaishnav) [reply _implementors/json_reference.md]: revereted.

File: _implementors/json_reference.md (hunk 2 above)
Comment (bhupendra-vaishnav) [reply _implementors/json_reference.md]: done

File: _implementors/json_reference.md (hunk 3 above)
Comment (bhupendra-vaishnav) [reply _implementors/json_reference.md]: Do you want to add `<br><br>` or `<br/><br/>` ?

File: _implementors/json_reference.md [hunk 5]
```diff
@@ -60,8 +61,8 @@ The focus of `devcontainer.json` is to describe how to enrich a container for th
...
+| `dockerComposeFile` | string,<br>array | **Required** when [using Docker Compose](https://docs.docker.com/compose/). Path or an ordered list of paths to Docker Compose files relative to the `devcontainer.json` file. Using an array is useful [when extending your Docker Compose configuration](https://docs.docker.com/compose/extends/#multiple-compose-files). The order of the array matters since the contents of later files can override values set in previous ones.<br>The default `.env` file is picked up from the root of the project, but you can use `env_file` in your Docker Compose file to specify an alternate location.<br>Note that the array syntax will execute the command without a shell. You can [learn more](#formatting-string-vs-array-properties) about formatting string vs array properties. |
+| `service` | string | **Required** when [using Docker Compose](https://docs.docker.com/compose/). The name of the service `devcontainer.json` supporting services / tools should connect to once running.  |
```
Comment (bhupendra-vaishnav) [inline _implementors/json_reference.md]: I am seeing some changes here for this link `[using Docker Compose]`. What should I do? Revert?

File: _implementors/json_reference.md [hunk 6]
```diff
@@ -76,12 +77,12 @@ When creating or working with a dev container, you may need different commands t
...
+| `postCreateCommand` 🏷️ | string,<br>array,<br>object | This command is the last of three that finalizes container setup when a dev container is created. It happens after `updateContentCommand` and once the dev container has been assigned to a user for the first time.<br><br />Cloud services can use this command to take advantage of user specific secrets and permissions.<br>Note that the array syntax will execute the command without a shell. You can [learn more](#formatting-string-vs-array-properties) about formatting string vs array vs object properties. |
+| `postStartCommand` 🏷️ | string,<br>array,<br>object | A command to run each time the container is successfully started.<br>Note that the array syntax will execute the command without a shell. You can [learn more](#formatting-string-vs-array-properties) about formatting string vs array vs object properties. |
```
Comment (bhupendra-vaishnav) [inline _implementors/json_reference.md]: I am seeing similar change for single-line-break HTML tag here. What should we do ? All revert to two single-line-break tag?

Also look at next line.

File: _implementors/json_reference.md (hunk 5 above)
Comment (samruddhikhandale) [reply _implementors/json_reference.md]: This looks good!

File: _implementors/json_reference.md (hunk 3 above)
Comment (samruddhikhandale) [reply _implementors/json_reference.md]: Let's keep the old changes intact (which were `<br><br>`)

File: _implementors/json_reference.md (hunk 6 above)
Comment (samruddhikhandale) [reply _implementors/json_reference.md]: Let's keep it as it is

File: _implementors/json_reference.md [hunk 7]
```diff
@@ -163,15 +162,15 @@ Finally, you may use an object format:
...
-| `${devcontainerId}` | Any | Identifier derived from a set of container labels that uniquely identify the dev container on a Docker host. It allows Features to refer to an identifier that is unique to the dev container they are installed into and that is stable across rebuilds.<br> The properties supporting it in devcontainer.json are: `name`, `runArgs`, `initializeCommand`, `onCreateCommand`, `updateContentCommand`, `postCreateCommand`, `postStartCommand`, `postAttachCommand`, `workspaceFolder`, `workspaceMount`, `mounts`, `containerEnv`, `remoteEnv`, `containerUser`, `remoteUser`, and `customizations`. |
+| `${devcontainerId}` | Any | Allow features to refer to an identifier that is unique to the dev container they are installed into and that is stable across rebuilds.<br> The properties supporting it in devcontainer.json are: `name`, `runArgs`, `initializeCommand`, `onCreateCommand`, `updateContentCommand`, `postCreateCommand`, `postStartCommand`, `postAttachCommand`, `workspaceFolder`, `workspaceMount`, `mounts`, `containerEnv`, `remoteEnv`, `containerUser`, `remoteUser`, and `customizations`. |
```
Comment (samruddhikhandale) [inline _implementors/json_reference.md]:
```suggestion
| `${devcontainerId}` | Any | Allow Features to refer to an identifier that is unique to the dev container they are installed into and that is stable across rebuilds.<br> The properties supporting it in devcontainer.json are: `name`, `runArgs`, `initializeCommand`, `onCreateCommand`, `updateContentCommand`, `postCreateCommand`, `postStartCommand`, `postAttachCommand`, `workspaceFolder`, `workspaceMount`, `mounts`, `containerEnv`, `remoteEnv`, `containerUser`, `remoteUser`, and `customizations`. |

```

I know you copied over this, but features should be with uppercase `F`

File: _implementors/json_reference.md (hunk 3 above)
Comment (bhupendra-vaishnav) [reply _implementors/json_reference.md]: done

Comment (samruddhikhandale) [reply _implementors/json_reference.md]: @bhupendra-vaishnav Did you miss to push the changes? I still see old diff with missing `<br>`

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: _implementors/json_reference.md (hunk 4 above)
Comment (bhupendra-vaishnav) [reply _implementors/json_reference.md]: done

File: _implementors/json_reference.md (hunk 3 above)
Comment (bhupendra-vaishnav) [reply _implementors/json_reference.md]: fixed and I re-validated. I think it should be good now.

**Comment by bhupendra-vaishnav** [comment]
@samruddhikhandale Would this be visible on the page? Look at blue-highlighted section in attached screen-shot : 


image

**Review by samruddhikhandale — APPROVED** [review]

**Pull request merged**